src/qtiCommonRenderer/renderers/interactions/TextEntryInteraction.js

```diff
@@ -132,7 +132,7 @@ var render = function render(interaction) {
 
             hideTooltip($input);
 
-            if ($input.val().length && regex.test($input.val())) {
+            if (regex.test($input.val())) {
```

Comment:
All right, now it's gonna check against a regex even en empty input. But I don't think that's what's intended to be done here.
What I said earlier about the test itself being possibly incorrect should be double checked with business, but I'm pretty sure we should not trigger the check against an empty field.

Reply:
@JimmyOAT

Reply:
This code is executed inside `else if (attributes.patternMask)` so I think it can be as you said, only check regex. For now, I have some errors in my local bosa environment, so I can't test properly

Reply:
I see, kindly re-request my review once you established the correct behavior, and checked it locally.
Thank you!

Reply:
Oh, wait. Never mind. The code was already causing the error popover to appear in case of an empty input, so your solution is totally fine.